Traceback (most recent call last):
  File "/local/295392/main.py", line 136, in <module>
    posterior_samples = metropolis_hasting(rng_key, max_samples, assumption_variance, Xu_noise, jnp.eye(2)*prior_var, param_iter, Xu_fixed, Xf, Y)
  File "/local/295392/include/mcmc_posterior.py", line 52, in metropolis_hasting
    cov_current = compute_K(init, xvague_sample_current, Xfz, Xfg)
  File "/local/295392/include/mcmc_posterior.py", line 17, in compute_K
    zz_uu = heat_equation_kuu(Xuz, Xuz, params)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 179, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/pjit.py", line 248, in cache_miss
    outs, out_flat, out_tree, args_flat, jaxpr, attrs_tracked = _python_pjit_helper(
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/pjit.py", line 136, in _python_pjit_helper
    infer_params_fn(*args, **kwargs)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/api.py", line 325, in infer_params
    return pjit.common_infer_params(pjit_info_args, *args, **kwargs)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/pjit.py", line 495, in common_infer_params
    jaxpr, consts, out_shardings, out_layouts_flat, attrs_tracked = _pjit_jaxpr(
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/pjit.py", line 1150, in _pjit_jaxpr
    jaxpr, final_consts, out_type, attrs_tracked = _create_pjit_jaxpr(
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/linear_util.py", line 350, in memoized_fun
    ans = call(fun, *args)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/pjit.py", line 1089, in _create_pjit_jaxpr
    jaxpr, global_out_avals, consts, attrs_tracked = pe.trace_to_jaxpr_dynamic(
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/profiler.py", line 336, in wrapper
    return func(*args, **kwargs)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/interpreters/partial_eval.py", line 2314, in trace_to_jaxpr_dynamic
    jaxpr, out_avals, consts, attrs_tracked = trace_to_subjaxpr_dynamic(
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/interpreters/partial_eval.py", line 2336, in trace_to_subjaxpr_dynamic
    ans = fun.call_wrapped(*in_tracers_)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/linear_util.py", line 192, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/local/295392/include/heat2d.py", line 82, in heat_equation_kuu
    return rbf_kernel(x1, x2, params)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 179, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/pjit.py", line 248, in cache_miss
    outs, out_flat, out_tree, args_flat, jaxpr, attrs_tracked = _python_pjit_helper(
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/pjit.py", line 136, in _python_pjit_helper
    infer_params_fn(*args, **kwargs)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/api.py", line 325, in infer_params
    return pjit.common_infer_params(pjit_info_args, *args, **kwargs)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/pjit.py", line 495, in common_infer_params
    jaxpr, consts, out_shardings, out_layouts_flat, attrs_tracked = _pjit_jaxpr(
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/pjit.py", line 1150, in _pjit_jaxpr
    jaxpr, final_consts, out_type, attrs_tracked = _create_pjit_jaxpr(
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/linear_util.py", line 350, in memoized_fun
    ans = call(fun, *args)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/pjit.py", line 1089, in _create_pjit_jaxpr
    jaxpr, global_out_avals, consts, attrs_tracked = pe.trace_to_jaxpr_dynamic(
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/profiler.py", line 336, in wrapper
    return func(*args, **kwargs)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/interpreters/partial_eval.py", line 2314, in trace_to_jaxpr_dynamic
    jaxpr, out_avals, consts, attrs_tracked = trace_to_subjaxpr_dynamic(
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/interpreters/partial_eval.py", line 2336, in trace_to_subjaxpr_dynamic
    ans = fun.call_wrapped(*in_tracers_)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/linear_util.py", line 192, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/local/295392/include/kernels.py", line 22, in rbf_kernel
    square_distances = jnp.sum((x1[:, jnp.newaxis, :] - x2[jnp.newaxis, :, :]) ** 2 / initial_theta['lengthscale'] ** 2,
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/numpy/array_methods.py", line 736, in op
    return getattr(self.aval, f"_{name}")(self, *args)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/numpy/array_methods.py", line 349, in _getitem
    return lax_numpy._rewriting_take(self, item)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 4590, in _rewriting_take
    return _gather(arr, treedef, static_idx, dynamic_idx, indices_are_sorted,
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 4599, in _gather
    indexer = _index_to_gather(shape(arr), idx)  # shared with _scatter_update
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 4707, in _index_to_gather
    idx = _canonicalize_tuple_index(len(x_shape), idx)
  File "/home/yanw/.conda/envs/uncertain/lib/python3.9/site-packages/jax/_src/numpy/lax_numpy.py", line 5027, in _canonicalize_tuple_index
    raise IndexError(
IndexError: Too many indices for array: 2 non-None/Ellipsis indices for dim 1.
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[17032,1],0]
  Exit code:    1
--------------------------------------------------------------------------
cp: missing destination file operand after '*.png/home/yanw/uncertain/results/figures/0624'
Try 'cp --help' for more information.
cp: missing destination file operand after '/home/yanw/uncertain'
Try 'cp --help' for more information.
mv: cannot stat '*.log': No such file or directory
